### PR TITLE
cdc: fix sarama case sensitivity

### LINKED ISSUE #100706: cdc: case insensitive sink config
See https://github.com/cockroachlabs/support/issues/2208

Since most of our WITH options are now case-insensitive, it's surprising when the sink JSON config isn't. For enums in particular like compression codecs, we should either be case insensitive or have better error messages that list all valid values. 

Jira issue: CRDB-26587

Epic CRDB-8667

### PR DESCRIPTION
**Pull request by HonoreDB**
Fixes #100706 by lowercasing compression codecs and passing them through transparently to Sarama rather than doing our own validation.

Release note (enterprise change): The kafka_sink_config Compression and RequiredAcks options are now case-insensitive.

### CODE REVIEW

File: pkg/ccl/changefeedccl/sink_kafka.go [hunk 1]
```diff
@@ -141,34 +142,23 @@ func (s *kafkaSink) getConcreteType() sinkType {
...
-	codec, err := validateCompressionCodec(s)
-	if err != nil {
-		return err
+func (j *compressionCodec) UnmarshalText(b []byte) error {
+	var c sarama.CompressionCodec
+	if err := c.UnmarshalText(bytes.ToLower(b)); err != nil {
+		return errors.WithHintf(err, "supported compression codecs are %s", getValidCompressionCodecs())
 	}
-	*j = compressionCodec(codec)
+	*j = compressionCodec(c)
```
Comment (miretskiy) [inline pkg/ccl/changefeedccl/sink_kafka.go:161]: how does that work? Isn't c always == none?

**Review by miretskiy — CHANGES_REQUESTED** [review]

**Review by HonoreDB — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/100929)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @jayshrivastava and @miretskiy)

___
*[`pkg/ccl/changefeedccl/sink_kafka.go` line 148 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/100929#gh-1160869046:-NSfoAky5Su1kO_ISs9E:b-sz4lrl) ([raw file](https://github.com/cockroachdb/cockroach/blob/71afbb8ce4925ab50efc32865890d51795f46b6d/pkg/ccl/changefeedccl/sink_kafka.go#L148)):*
<details><summary><i>Previously, miretskiy (Yevgeniy Miretskiy) wrote…</i></summary><blockquote>

can we not do this?  iterate until we get out of bounds error
and then catch it... Just not necessary.

I don't understand why we can't just call toUpper when doing validation.

If you are worried about upgrading sarama (and it claims to support new codec that we don't know about), then just write a test -- basically what you have here, that goes through every 
compression code and makes sure that we have thhat codec in the saramCompressionCodecOptions.

</blockquote></details>

Done. That was indeed the worry, I've moved it to a test.
___
*[`pkg/ccl/changefeedccl/sink_kafka.go` line 161 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/100929#gh-1160869660:-NSfoH_jAwX87PyvnhUb:b-ady1zu) ([raw file](https://github.com/cockroachdb/cockroach/blob/71afbb8ce4925ab50efc32865890d51795f46b6d/pkg/ccl/changefeedccl/sink_kafka.go#L161)):*
<details><summary><i>Previously, miretskiy (Yevgeniy Miretskiy) wrote…</i></summary><blockquote>

how does that work? Isn't c always == none?
</blockquote></details>

c.UnmarshalText is a pointer method that sets c.

**Review by miretskiy — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/100929)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @jayshrivastava)

**Comment by HonoreDB** [comment]
bors r=[miretskiy]

**Pull request merged**